x-pack/plugins/security_solution/server/endpoint/routes/metadata/index.ts

```diff
@@ -103,12 +108,24 @@ export function registerEndpointRoutes(router: IRouter, endpointAppContext: Endp
           context.core.savedObjects.client
         );
 
+        const statusIDs = req.body?.agent_status?.length
+          ? await findAgentIDsByStatus(
+              agentService,
+              context.core.savedObjects.client,
+              req.body?.agent_status
+            )
+          : null;
+        if (statusIDs === undefined) {
+          return res.badRequest({ body: 'invalid status' });
+        }
+
```

Comment:
I think we have to take a step back and consider how we want to implement this. What is the default when it is empty? And does the filter indicate inclusion or exclusion? Regardless I think we want to ask ingest for agents not in a target status list so that we can just add the Agent ID to the must not list when we query for endpoints.

Reply:
if `agent_status` is not requested (is that what you mean by empty?) then it does not filter by status. `statusIDs` will be `null`.

The filter is exclusive to the requested statuses. It seemed intuitive to me. `agent_status: ["online"]` will return only agents who have the status `online`. if you have a KQL query or something, it wont be the union "KQL query results + online agents". It will be the intersection. Which is pretty default for filtering by multiple things.

As for asking for agents _not_ in a target list, that is complicated. the 'online' kuery is already defined as: "NOT offline AND NOT error AND NOT enrolling AND NOT unenrolling".  So, sure, we could say "offline OR error OR enrolling OR unenrolling" and then say `must_not  ... those IDs`. 

if the incoming query is then `agent_status: ["online", "enrolling"]` then the invert we would just remove 'enrolling' from that group. but if  'online' is removed from the request, then it's a totally different ballgame. We would have to self-build the query, and hope ingest never changes how they define any of those statuses. if we reuse the provided builders, our query code is not so brittle

Reply:
Because of the state of the application and the fact that we only currently know and care about a subset of ingest status; we want to find the agents not in the status we want and exclude them. This is how we do it today for unenrolled. Must not is filter context and so it is faster too. This may or may not change when we move to a single document per host index.